crates/block-producer/src/block_builder/mod.rs

```diff
@@ -230,13 +231,32 @@ impl BlockBuilder {
         proposed_block: ProposedBlock,
         block_inputs: BlockInputs,
     ) -> Result<(OrderedBatches, BlockInputs, BlockHeader, BlockBody), BuildBlockError> {
-        let response = self
-            .validator
-            .validate_block(proposed_block.clone())
-            .await
-            .map_err(BuildBlockError::ValidateBlockFailed)?;
+        // Request validation concurrently to building the block ourselves.
+        let (validator_result, build_result) = tokio::join!(
+            // Send request to validator.
+            self.validator.validate_block(proposed_block.clone()),
+            // Build the block ourselves.
+            tokio::task::spawn_blocking({
+                let proposed_block = proposed_block.clone();
+                move || build_block(proposed_block)
+            })
+        );
```

Comment:
Could we not do this without explicitly spawning new threads - something like this:
```rust
let response = self.validator.validate_block(proposed_block.clone());
let local_block = build_block(proposed_block);
let response = response.await.map_err(BuildBlockError::ValidateBlockFailed)?;
```

Or is this undesirable for some reason?

Reply:
Small caveat: The default blocking threadpool is _very_ large, which can lead to starvation and running into timeouts up the call-chain.